src/router/task.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const Task = require('../models/task')
+const router = new express.Router()
+
+router.post('/', async (req, res) => {
+    const todo = new Task(req.body)
+    todo.save().then(() => {
+        res.send(todo)
+    }).catch((e) => {
+        res.send(e)
+    })
+})
+
+router.get('/', async (req, res) => {
+    Task.find({}).then((users) => {
+        res.send(users)
+        }).catch((e) => {
+        res.status(400).send(e)
+    })
+})
+
+router.get('/:id', async (req, res) => {
+    try {
+        const task = await Task.findOne({ _id: req.params.id })
+
+        if (!task) {
+            return res.status(404).send()
+        }
+        res.send(task)
+    } catch (e) {
```

Comment:
google if express can do it for you. Decouple if not.

Reply:
Task.findById({_id: req.params.id}, (err, doc) => {
        if (err) {
            res.status(404).send(err)
        }
        res.send(doc)
    })

Reply:
still bad, too much of code that's not related to the method business logic.